definitions/checks/system_registration.rb

```diff
@@ -5,16 +5,29 @@ class Checks::SystemRegistration < ForemanMaintain::Check
     tags :default
 
     confine do
-      file_exists?('/etc/rhsm/rhsm.conf') &&
-        !feature(:foreman_server) &&
-        feature(:foreman_proxy)
+      # FIXME: Once this https://projects.theforeman.org/issues/25549 issue
+      # gets fix then please fix code here. Only feature(:downstream) expected.
+      (feature(:downstream) || (!feature(:katello) && feature(:foreman_proxy))) &&
```

Comment:
@kgaikwad it seems it does not exclude upstream proxy. Wouldn't something like `feature(:downstream) || (execute('rpm -q foreman-proxy') =~ /sat.noarch/)` describe better the options?
Thanks for the #FIXME comment!